src/search.h

```diff
@@ -156,16 +156,56 @@ class ISearchManager {
     virtual void check_time(Search::Worker&) = 0;
 };
 
+struct InfoShort {
+    int         depth;
+    std::string score;
+};
+
+struct InfoFull: InfoShort {
```

Comment:
I'm a bit torn on using inheritance here. I guess we'll see it if causes any maintainability issues

Reply:
Yeah I didn't like this either much, ideally I dont want to even expose 3 functions, rather 2. 
The only reason for the third one is this...
```cpp
    if (rootMoves.empty())
    {
        rootMoves.emplace_back(Move::none());
        main_manager()->updates.onUpdateShort(
          {0, UCIEngine::to_score(rootPos.checkers() ? -VALUE_MATE : VALUE_DRAW, rootPos)});
    }
```
perhaps this is not even the right place.. we should have a check earlier if the position has no moves and not check this after starting a search thread?

Reply:
We could check this somewhere else, but the interface is still needed. This situation must be handlable by search. Making the fields optional would be unreasonable. If it's only used here then perhaps the info should be named InfoTerminalRoot or something along these lines.